### ISSUE TITLE
Unable to generate models for meta jsonschema

### ISSUE THREAD

**Issue by Skeen** [koxudaxi/datamodel-code-generator#447]
**Describe the bug**
I expected to be able to generate Pydanic models for validating JSONSchemas themselves.

**To Reproduce**
Fetch the json-schema meta-schema draft 7:
```
wget http://json-schema.org/draft-07/schema
```
Run datamodel-codegen:
```
datamodel-codegen --input schema --input-file-type jsonschema
```

**Expected behavior**
I expected data-models able to parse JSONSchemas to be generated.

**Version:**
 - OS: [e.g. iOS] Ubuntu 20.04.2 LTS
 - Python version: Python 3.8.5
 - datamodel-code-generator version: [e.g. 22] 0.11.7

**Comment by koxudaxi** [comment]
@Skeen 
Thank you for creating this issue.
I have confirmed the error.
But, I don't understand why the error occurs.
I will check it.

**Comment by koxudaxi** [comment]
@Skeen 

I ran datamodel-code-generator for the schema 
output:
```
pydantic.error_wrappers.ValidationError: 6 validation errors for JsonSchemaObject
properties -> default
  value is not a valid dict (type=type_error.dict)
properties -> examples -> items
  value is not a valid list (type=type_error.list)
properties -> examples -> items
  value is not a valid dict (type=type_error.dict)
properties -> const
  value is not a valid dict (type=type_error.dict)
properties -> enum -> items
  value is not a valid list (type=type_error.list)
properties -> enum -> items
  value is not a valid dict (type=type_error.dict)
```

I check the detail of the schema.
```
        "examples": {
            "type": "array",
            "items": true
        },
```

I don't understand why `items` have `true`. I expect a list of the object.

What do you think about it?

**Comment by Skeen** [comment]
> ```
>         "examples": {
>             "type": "array",
>             "items": true
>         },
> ```
> 
> I don't understand why `items` have `true`. I expect a list of the object.
> 
> What do you think about it?

This is unexpected to me too, I'd have imagined either a type definition or a reference within an object, not just `true`.

**Comment by xkortex** [comment]
I was toying around with this issue and it seems like the main problem is that there are several fields (e.g. `additionalItems`, `additionalProperties`, `items`, maybe some others) which are set to `true` in the jsonschema draft, but the `JsonSchemaObject` isn't really able to take this into account. I think maybe just allowing for these fields to be `Union[whatever, bool]` would get through snag. 

Would be nice if it could self-bootstrap the jsonschema spec itself :)

**Comment by xkortex** [comment]
Possibly relevant: https://github.com/OAI/OpenAPI-Specification/issues/668#issue-150416089

So I think we might be able to substitute in the empty object in places where we encounter these `true` fields.

Also I was thinking about the whole self-hosting thing a bit more. In theory, since the json-schema describes itself, we ought to be able to compile the json-schema draft (e.g. draft-7.json) to python, then use that to parse that very same draft. The output of that parse should compile back to python. This could be a very useful way to test the veracity of the parser and generator.

In fact, that "self-compiling" python code could act as the main parser in the library. It's much, much shorter, so I wonder why there is so much manual logic in the current JsonSchemaObject. Maybe that's the bootstrapping overhead? :)

**Comment by tbsexton** [comment]
I was checking out that thread and i[t looks like, yep](https://github.com/OAI/OpenAPI-Specification/issues/668#issuecomment-593549261), the jsonschema spec treats `item: true` and `item: {}` identically. That's super interesting, and would solve [my use case](https://github.com/jsongraph/json-graph-specification/issues/55) if taken into account. 

I also think your idea about the self-describing schema is excellent; maybe worth a separate issue?

**Comment by tbsexton** [comment]
Also I believe this would solve #696

**Comment by sebastroy** [comment]
> @Skeen
> 
> I ran datamodel-code-generator for the schema output:
> 
> ```
> pydantic.error_wrappers.ValidationError: 6 validation errors for JsonSchemaObject
> properties -> default
>   value is not a valid dict (type=type_error.dict)
> properties -> examples -> items
>   value is not a valid list (type=type_error.list)
> properties -> examples -> items
>   value is not a valid dict (type=type_error.dict)
> properties -> const
>   value is not a valid dict (type=type_error.dict)
> properties -> enum -> items
>   value is not a valid list (type=type_error.list)
> properties -> enum -> items
>   value is not a valid dict (type=type_error.dict)
> ```
> 
> I check the detail of the schema.
> 
> ```
>         "examples": {
>             "type": "array",
>             "items": true
>         },
> ```
> 
> I don't understand why `items` have `true`. I expect a list of the object.
> 
> What do you think about it?

My understanding of the v7 specification is that array with items "true" are tuples, not lists where additional items are accepted (would false otherwise) see https://json-schema.org/understanding-json-schema/reference/array.html#items under the tuple section.

**Comment by koxudaxi** [comment]
I'm sorry for my too-late reply.
I have released the PR as `0.15.0`
Thank you very much!!

**Comment by sebastroy** [comment]
Thank you for all the good work!
